[Swift in WebKit] Add Swift Testing support to WebKit

rr-codes opened the pull request:
#### 0abae04c2bcebc137e640a232b2d8794fc89ab3d
<pre>
[Swift in WebKit] Add Swift Testing support to WebKit
<a href="https://bugs.webkit.org/show_bug.cgi?id=283012">https://bugs.webkit.org/show_bug.cgi?id=283012</a>
<a href="https://rdar.apple.com/139627749">rdar://139627749</a>

Reviewed by Wenson Hsieh.

Support writing tests using Swift Testing.

A few things should be noted:

* This currently only works with the internal SDK

* Eventually, there will need to be solution for EWS since there&apos;s no way to make a single test plan have different paths depending on the internal sdk or not.

* Tools/TestWebKitAPI/Configurations/TestWebKitAPILibrary.xcconfig:
* Tools/TestWebKitAPI/TestBundle/GoogleTests.mm:
(+[GoogleTestLoader registerTestClasses]):
* Tools/TestWebKitAPI/TestBundle/TestWebKitAPI.xctestplan:
* Tools/TestWebKitAPI/TestWebKitAPI.xcodeproj/project.pbxproj:
* Tools/TestWebKitAPI/Tests/WebKit Swift/TestWebKitAPIBundle-Bridging-Header.h: Added.
* Tools/TestWebKitAPI/Tests/WebKit Swift/WKWebViewSwiftOverlay.swift: Added.
(WKWebViewSwiftOverlayTests.evaluateJavaScriptYieldsExpectedResponse):
(WKWebViewSwiftOverlayTests.evaluateJavaScriptWithNilResponse):

Canonical link: <a href="https://commits.webkit.org/286626@main">https://commits.webkit.org/286626@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/dbaa05f5a99c079d440f67c6422db2e3f42edb69

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/76413 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/55448 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/29319 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/80936 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/27689 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/78530 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/64590 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/3741 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/59908 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/18027 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/79480 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/49822 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/65617 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/40239 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/47217 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/23106 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/26011 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/68343 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/23438 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/82386 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/3789 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/2478 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/68198 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/3942 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/65586 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/67436 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/16851 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/11402 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/9503 "Passed tests") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/3736 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/3759 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/7189 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/5517 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

whsieh commented on Tools/TestWebKitAPI/TestBundle/TestWebKitAPI.xctestplan:
```diff
@@ -14,7 +14,7 @@
     {
       "target" : {
-        "containerPath" : "container:TestWebKitAPI.xcodeproj",
+        "containerPath" : "container:..\/OpenSource\/Tools\/TestWebKitAPI\/TestWebKitAPI.xcodeproj",
```
It's a bit surprising to see a relative path like this in open source code. This distinction only makes sense in the context of compiling WebKit/Safari using the Apple internal SDK.

pxlcoder replied:
+1 this doesn't seem right

pxlcoder commented on Tools/TestWebKitAPI/TestBundle/GoogleTests.mm:
```diff
@@ -233,7 +233,16 @@ + (void)registerTestClasses
+            ]];
+
+            if (![excludedClassesFromAssertion containsObject:className])
+                NSAssert(added, @"Failed to add Google Test method \"%@\", this method may already exist in the class.", methodName);
```
Was this assertion being hit before your change? If so, this seems like it should be made separately?

Also FIXME should have a bug associated with it.

rr-codes replied:
yes, it was; the test plan was just not being executed by anyone so it went unnoticed. I figured it's ok to make it in this change since it's all test-only code anyways, and debug-only code at that, and isn't even really a "bug" per se since there's no functionality impacted really

pxlcoder replied:
It would just be unfortunate if the real change (everything other than this) had to get reverted due to breaking some builds, and we lose this fix too 😅

rr-codes committed:
[Swift in WebKit] Add Swift Testing support to WebKit
https://bugs.webkit.org/show_bug.cgi?id=283012
rdar://139627749

Reviewed by Wenson Hsieh.

Support writing tests using Swift Testing.

A few things should be noted:

* This currently only works with the internal SDK

* Eventually, there will need to be solution for EWS since there's no way to make a single test plan have different paths depending on the internal sdk or not.

* Tools/TestWebKitAPI/Configurations/TestWebKitAPILibrary.xcconfig:
* Tools/TestWebKitAPI/TestBundle/GoogleTests.mm:
(+[GoogleTestLoader registerTestClasses]):
* Tools/TestWebKitAPI/TestBundle/TestWebKitAPI.xctestplan:
* Tools/TestWebKitAPI/TestWebKitAPI.xcodeproj/project.pbxproj:
* Tools/TestWebKitAPI/Tests/WebKit Swift/TestWebKitAPIBundle-Bridging-Header.h: Added.
* Tools/TestWebKitAPI/Tests/WebKit Swift/WKWebViewSwiftOverlay.swift: Added.
(WKWebViewSwiftOverlayTests.evaluateJavaScriptYieldsExpectedResponse):
(WKWebViewSwiftOverlayTests.evaluateJavaScriptWithNilResponse):

Canonical link: https://commits.webkit.org/286626@main
